### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by gnavi07**
Please review the scala code of finding top N movies based on avg rating

### CODE REVIEW

File: Problem2.scala [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  
+  val sparkConf = new SparkConf()
+  sparkConf.set("spark.app.Name","my app")
+  sparkConf.set("spark.master","local[2]")
+  
+  val spark = SparkSession.builder()
+  .config(sparkConf)
+  .getOrCreate()
+  
+  val ratingsRdd = spark.sparkContext.textFile("/Users/smili/Documents/Big data -Naveena/Week 12/dataset for assignment/ratings.dat")
```
Comment (pcbzmani) [inline Problem2.scala]: Make path name as input file path

`val ratingsRdd = spark.sparkContext.textFile("input file path/ratings.dat")`

File: Problem2.scala [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import org.apache.spark.SparkConf
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.functions._
+
+
+
+object Problem2 extends App{
+  
+  val sparkConf = new SparkConf()
+  sparkConf.set("spark.app.Name","my app")
```
Comment (pcbzmani) [inline Problem2.scala]: App name as Movie rating

File: Problem2.scala [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import org.apache.spark.SparkConf
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.functions._
+
+
+
+object Problem2 extends App{
```
Comment (pcbzmani) [inline Problem2.scala]: Object name as movie rating

File: Problem2.scala [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+      (fields(0),fields(1),fields(2),fields(3))
+    })
+    
+    import spark.implicits._
+    
+    val ratingsDf=spark.createDataFrame(ratingMap)
+    .toDF("movieid","custid","rating","phoneno")
+    
+    
+      val moviesRdd = spark.sparkContext.textFile("/Users/smili/Documents/Big data -Naveena/Week 12/dataset for assignment/movies.dat")
```
Comment (pcbzmani) [inline Problem2.scala]: file name as input file path

`val moviesRdd = spark.sparkContext.textFile("/Users/smili/Documents/Big data -Naveena/Week 12/dataset for assignment/movies.dat")`
also check alignment

File: Problem2.scala [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    
+    val joinCond = moviesDf.col("movieid") === ratingsDf.col("movieid")
+    val joinType = "left"
+
+    val joinedDf=moviesDf.join(broadcast(ratingsDf),joinCond,joinType)
+    .drop(ratingsDf.col("movieid"))
+    
+    joinedDf.createOrReplaceTempView("Movies")
+    val sumRatDf=spark.sql("select movieid,moviename,sum(rating)/count(*) as Avg from movies group by movieid,moviename order by Avg desc")
+    .show(false)
```
Comment (pcbzmani) [inline Problem2.scala]: Display the results as commented.

**Review by pcbzmani — CHANGES_REQUESTED** [review]
Please resolve all comments

File: Problem2.scala (hunk 1 above)
Comment (gnavi07) [reply Problem2.scala]: file-path has been updated to <input file path>

File: Problem2.scala (hunk 3 above)
Comment (gnavi07) [reply Problem2.scala]: object name made to topMovies instead of Problem2

File: Problem2.scala (hunk 5 above)
Comment (gnavi07) [reply Problem2.scala]: Results are commented and displayed

File: Problem2.scala (hunk 2 above)
Comment (gnavi07) [reply Problem2.scala]: App name is changed to Movie rating

**Review by pcbzmani — APPROVED** [review]
Approve

**Pull request merged**